examples/language-modeling/run_lora_clm.py

```diff
@@ -676,19 +700,25 @@ def main():
     if tokenizer.pad_token_id is None:
         tokenizer.pad_token_id = tokenizer.eos_token_id
 
-    def tokenize(prompt, add_eos_token=True):
+    def tokenize(prompt, add_eos_token=True, add_bos_token=True):
+        add_eos_token_o = tokenizer.add_eos_token
+        add_bos_token_o = tokenizer.add_bos_token
         if not data_args.dataset_concatenation:
-            add_eos_token = False
+            tokenizer.add_eos_token = add_eos_token
```

Comment:
@sywangyi what about the else condition? Does `tokenizer.add_eos_token` exist?

Reply:
the model tokenizer class contain such information. like https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/tokenization_llama_fast.py#L135-L136  and I do not want to override this value. since when dataset_concatenation is on, we will add eos manually, we only need tokenizer to add eos when dataset_concatenation is off.